external/fv3fit/tests/training/test_main.py

```diff
@@ -86,86 +105,135 @@ def mock_load_batches():
         yield magic_load_mock
 
 
+def call_main(
+    tmpdir, mock_load_batches, additional_variables, derived_output_variables,
+):
+    model_type = "DenseModel"
+    hyperparameters = fv3fit.DenseHyperparameters(depth=2, width=8)
+    config = get_config(
+        tmpdir,
+        additional_variables,
+        derived_output_variables,
+        model_type,
+        hyperparameters,
+    )
+    mock_load_batches.return_value = [config.mock_dataset for _ in range(6)]
+    all_variables = (
+        config.input_variables + config.output_variables + additional_variables
+    )
+    with mock.patch("fv3fit.DerivedModel") as MockDerivedModel:
+        MockDerivedModel.return_value = mock.MagicMock(
+            name="derived_model_return", spec=fv3fit.Predictor
+        )
+        fv3fit.train.main(config.args)
+    return CallArtifacts(
+        config.output_path,
+        all_variables,
+        MockDerivedModel,
+        config.input_variables,
+        config.output_variables,
+        hyperparameters,
+    )
+
+
 @pytest.mark.parametrize("additional_variables", [[], ["downward_shortwave"]])
-def test_main(
-    mock_dataset: xr.Dataset,
+@pytest.mark.parametrize("derived_output_variables", [[], ["downwelling_shortwave"]])
```

Comment:
Is the parameterization of `derived_output_variables` relevant to this test?

Reply:
It could be, if load_batches is only called correctly depending on whether derived output variables are requested. If it is it's a bug, but that's why the test is here.